sklearnex/tests/test_n_jobs_support.py

```diff
@@ -79,15 +79,27 @@ def check_methods_decoration(estimator):
     check_estimator_doc(estimator_instance)
     # check `n_jobs` log entry for supported methods
     # `fit` call is required before other methods
-    check_method(X, Y, method=estimator_instance.fit, caplog=caplog)
+    # `partial_fit` is checked last if necessary
+    if "y" in inspect.signature(estimator_instance.fit).parameters:
+        check_method(X, y, method=estimator_instance.fit, caplog=caplog)
+    else:
+        check_method(X, method=estimator_instance.fit, caplog=caplog)
```

Comment:
because fit method does not necessarily contain y (for example, it's true for covariance estimator)

Reply:
![image](https://github.com/intel/scikit-learn-intelex/assets/24354279/1169df68-f93b-43e0-b59f-6924abcaa291)
It's API mistake in IncCov estimator. All sklearn-like estimators should list `y` as parameter.

Reply:
![image](https://github.com/intel/scikit-learn-intelex/assets/24354279/d84fc37b-5d46-43ac-9449-33d69dabbfb9)

Reply:
you are right, thanks for pointing out. your suggestion does not work though because not all estimators have partial_fit method. but I removed possibility of fit call without y. please check if the logic is fine now.

Reply:
This issue will be corrected in this PR as well, #1737 which does it based on the number of arguments without default values.